### ISSUE TITLE
Severity of implicitly_unwrapped_optional can’t be changed to error

### ISSUE THREAD

**Issue by lo1tuma** [realm/SwiftLint#4661]
I’ve configured the `implicitly_unwrapped_optional` rule to report violations as an error, but I still get them reported as a warning:

Config:
```yml
implicitly_unwrapped_optional: error
```

Emoji reporter output:
```
⚠️ Line 48: Implicitly unwrapped optionals should be avoided when possible.
```

**Comment by lo1tuma** [comment]
I just found the problem on my side. I had a configuration error. It seems like specifying the severity like this `implicitly_unwrapped_optional: error` is not possible because there is also a `mode` option. I thought the shorthand severity configuration should be always possible for all rules, but it seems like that assumption is wrong.

I’ve then tried to use:
```yml
implicitly_unwrapped_optional:
  severity: error
   mode: allExceptIBOutlets
```

because that is the information I got from the [docs](https://realm.github.io/SwiftLint/implicitly_unwrapped_optional.html)

But the expected value for `mode` is different, it should be `all_except_iboutlets`, which I only found out by reading the source code of the rule.

Since I’m new to SwiftLint I would now have a few follow-up questions:
* Is there a way to validate the whole config to verify there are no errors? If not, is this already considered or shall I open a new feature request?
* Is there any documentation that shows how the actual config properties and values should be named and maybe even explain what they mean?
* Is there any easy way to figure out which rules are binary rules and which have multiple options?
